### PR TITLE
Using new nbsite syntax and new doit command

### PR DESCRIPTION
**Pull request by jsignell**
* Switched to new nbsite syntax for generate_rst and build
* Added data_setup command to dodo
* Using new data_setup command on travis, appveyor
* Stopped using `ioam` channel

Spot tested locally that gallery works:
screen shot 2019-01-29 at 10 44 10 am

And json blob is copied (tried dragging slider):
screen shot 2019-01-29 at 10 45 09 am

Closes: #291

### CODE REVIEW

**Comment by philippjfr** [comment]
Just tagged a dev release, will see how that works out.

**Comment by philippjfr** [comment]
Looks like it failed when deploying the docs:

```
Deploying application
Initialized empty Git repository in /tmp/d20190129-8039-186hf2p/work/.git/
Switched to a new branch 'gh-pages'
cd -
cd /tmp/d20190129-8039-186hf2p/work
rsync: change_dir "/home/travis/build/pyviz/geoviews/doc/_build/html" failed: No such file or directory (2)
rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1183) [sender=3.1.0]
Could not copy /home/travis/build/pyviz/geoviews/doc/_build/html.
failed to deploy
```

I'll update the ``local_dir`` in the deploy section.

**Comment by jsignell** [comment]
I think I misunderstood what I was checking for when running the gallery script. I thought the screenshots were being dynamically created.

**Comment by philippjfr** [comment]
>I think I misunderstood what I was checking for when running the gallery script. I thought the screenshots were being dynamically created.

They are, the thumbnails are created correctly. The issue is that the notebooks themselves aren't executed so the output is just the raw code.

**Comment by jsignell** [comment]
The evaluated notebooks need to be in doc/gallery/

**Comment by philippjfr** [comment]
>The evaluated notebooks need to be in doc/gallery/

I don't follow, sphinx should evaluate the notebooks, the problem here is that the gallery script probably copies the notebooks to doc/gallery and nbsite then assumes that they have already been evaluated.

**Comment by jsignell** [comment]
If you want to run each notebook twice, then you could just delete the .ipynb files in doc/gallery after the thumbnailing has completed.

**Comment by philippjfr** [comment]
>If you want to run each notebook twice, then you could just delete the .ipynb files in doc/gallery after the thumbnailing has completed.

The thumbnailing does not execute the notebook it converts to script and runs that, I think the original intention of copying was that the cleared notebooks would be included in the build so they can be downloaded.

**Comment by jsignell** [comment]
Right then all you need is to delete the .ipynb files from /doc/gallery after the thumnailing finishes. The next command will build them.

**Comment by philippjfr** [comment]
>Right then all you need is to delete the .ipynb files from /doc/gallery after the thumnailing finishes.

Sure that's an easy fix now, since I'm guessing they are not included anyway, but the idea here is that they are eventually shipped as part of the docbuild so a user can download the notebook from the website.

File: .travis.yml [hunk 1]
```diff
@@ -74,29 +73,27 @@ jobs:
...
-        - cd doc
-        - nbsite_nbpagebuild.py ioam geoviews ../examples .
+        - doit data_setup
+        ### generate rst skeleton
+        - nbsite generate-rst --org pyviz --project-name geoviews --offset 0
         ### gallery
+        - cd doc
         - ln -s ../examples/assets assets
         - ln -s ../examples/data data
         - nbsite_gallery.py .. examples None '.' bokeh matplotlib None
```
Comment (jsignell) [inline .travis.yml]:
```suggestion
         - nbsite_gallery.py .. examples None '.' bokeh matplotlib None
         - rm gallery/**/*.ipynb
```

**Comment by philippjfr** [comment]
Thanks, should work now.

**Comment by philippjfr** [comment]
Superseded by #298

**Pull request closed without merging**